{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os, sys, glob\n",
    "import math\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'svg'\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_snv = pd.read_csv(\"raw/SNVs.csv\", low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>VAF</th>\n",
       "      <th>TCN</th>\n",
       "      <th>LCN</th>\n",
       "      <th>CCF</th>\n",
       "      <th>CLONALITY</th>\n",
       "      <th>Total_Index_Lesions_In_Patient</th>\n",
       "      <th>Num_Other_Mets_In_Patient</th>\n",
       "      <th>CLONALITY_In_Other_Mets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>105246551</td>\n",
       "      <td>105246551</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FOXA1</td>\n",
       "      <td>3169</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>38061240</td>\n",
       "      <td>38061240</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.298731257</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>CLONAL</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FOXA1</td>\n",
       "      <td>3169</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>38061240</td>\n",
       "      <td>38061240</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.139386189</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.961</td>\n",
       "      <td>CLONAL</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CLONAL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>23405</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>95574698</td>\n",
       "      <td>95574698</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>105246551</td>\n",
       "      <td>105246551</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0        AKT1             207  MSKCC     GRCh37         14       105246551   \n",
       "1       FOXA1            3169  MSKCC     GRCh37         14        38061240   \n",
       "2       FOXA1            3169  MSKCC     GRCh37         14        38061240   \n",
       "3      DICER1           23405  MSKCC     GRCh37         14        95574698   \n",
       "4        AKT1             207  MSKCC     GRCh37         14       105246551   \n",
       "\n",
       "   End_Position Strand Variant_Classification Variant_Type  ... HIGH_INF_POS  \\\n",
       "0     105246551      +      Missense_Mutation          SNP  ...          NaN   \n",
       "1      38061240      +      Missense_Mutation          SNP  ...          NaN   \n",
       "2      38061240      +      Missense_Mutation          SNP  ...          NaN   \n",
       "3      95574698      +      Missense_Mutation          SNP  ...          NaN   \n",
       "4     105246551      +      Missense_Mutation          SNP  ...          NaN   \n",
       "\n",
       "  MOTIF_SCORE_CHANGE          VAF  TCN  LCN    CCF CLONALITY  \\\n",
       "0                NaN          NaN  NaN  NaN    NaN       NaN   \n",
       "1                NaN  0.298731257  2.0  1.0  1.000    CLONAL   \n",
       "2                NaN  0.139386189  2.0  1.0  0.961    CLONAL   \n",
       "3                NaN          NaN  NaN  NaN    NaN       NaN   \n",
       "4                NaN          NaN  NaN  NaN    NaN       NaN   \n",
       "\n",
       "  Total_Index_Lesions_In_Patient Num_Other_Mets_In_Patient  \\\n",
       "0                            NaN                       NaN   \n",
       "1                            1.0                       1.0   \n",
       "2                            1.0                       1.0   \n",
       "3                            NaN                       NaN   \n",
       "4                            NaN                       NaN   \n",
       "\n",
       "   CLONALITY_In_Other_Mets  \n",
       "0                      NaN  \n",
       "1                   CLONAL  \n",
       "2                   CLONAL  \n",
       "3                      NaN  \n",
       "4                      NaN  \n",
       "\n",
       "[5 rows x 96 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_snv.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate patient to samples dict\n",
    "patientToSamplesDict = {}\n",
    "for sample in set(df_snv.Tumor_Sample_Barcode):\n",
    "    patient = \"-\".join(sample.split('-')[0:2])\n",
    "    if patient not in patientToSamplesDict:\n",
    "        patientToSamplesDict[patient] = set()\n",
    "    patientToSamplesDict[patient].add(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_cna = pd.read_csv(\"raw/CNAs.csv\", low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>P-0004434-T01-IM5</th>\n",
       "      <th>P-0004314-T02-IM6</th>\n",
       "      <th>P-0004216-T01-IM5</th>\n",
       "      <th>P-0011462-T01-IM5</th>\n",
       "      <th>P-0000464-T01-IM3</th>\n",
       "      <th>P-0012362-T01-IM5</th>\n",
       "      <th>P-0009740-T01-IM5</th>\n",
       "      <th>P-0008538-T01-IM5</th>\n",
       "      <th>P-0001952-T01-IM3</th>\n",
       "      <th>...</th>\n",
       "      <th>P-0014854-T01-IM6</th>\n",
       "      <th>P-0006376-T03-IM6</th>\n",
       "      <th>P-0003224-T01-IM5</th>\n",
       "      <th>P-0000473-T01-IM3</th>\n",
       "      <th>P-0011072-T01-IM5</th>\n",
       "      <th>P-0009444-T02-IM6</th>\n",
       "      <th>P-0016472-T01-IM6</th>\n",
       "      <th>P-0017362-T01-IM6</th>\n",
       "      <th>P-0015582-T01-IM6</th>\n",
       "      <th>P-0014692-T01-IM6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RB1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>-1.5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MDC1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PREX2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SOX17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1919 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  P-0004434-T01-IM5  P-0004314-T02-IM6  P-0004216-T01-IM5  \\\n",
       "0         RB1                  0                  0                  0   \n",
       "1        MDC1                  0                  0                  0   \n",
       "2        ARAF                  0                  0                  0   \n",
       "3       PREX2                  0                  0                  0   \n",
       "4       SOX17                  0                  0                  0   \n",
       "\n",
       "   P-0011462-T01-IM5  P-0000464-T01-IM3  P-0012362-T01-IM5  P-0009740-T01-IM5  \\\n",
       "0                  0                  0                  0               -1.5   \n",
       "1                  0                  0                  0                0.0   \n",
       "2                  0                  0                  0                0.0   \n",
       "3                  0                  0                  0                0.0   \n",
       "4                  0                  0                  0                0.0   \n",
       "\n",
       "   P-0008538-T01-IM5  P-0001952-T01-IM3  ...  P-0014854-T01-IM6  \\\n",
       "0                  0                  0  ...                  0   \n",
       "1                  0                  0  ...                  0   \n",
       "2                  0                  0  ...                  0   \n",
       "3                  0                  0  ...                  0   \n",
       "4                  0                  0  ...                  0   \n",
       "\n",
       "   P-0006376-T03-IM6  P-0003224-T01-IM5  P-0000473-T01-IM3  P-0011072-T01-IM5  \\\n",
       "0                  0                  0                  0                  0   \n",
       "1                  0                  0                  0                  0   \n",
       "2                  0                  0                  0                  0   \n",
       "3                  0                  0                  0                  0   \n",
       "4                  0                  0                  0                  0   \n",
       "\n",
       "   P-0009444-T02-IM6  P-0016472-T01-IM6  P-0017362-T01-IM6  P-0015582-T01-IM6  \\\n",
       "0                  0                  0                  0                  0   \n",
       "1                  0                  0                  0                  0   \n",
       "2                  0                  0                  0                  0   \n",
       "3                  0                  0                  0                  0   \n",
       "4                  0                  0                  0                  0   \n",
       "\n",
       "   P-0014692-T01-IM6  \n",
       "0                  0  \n",
       "1                  0  \n",
       "2                  0  \n",
       "3                  0  \n",
       "4                  0  \n",
       "\n",
       "[5 rows x 1919 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_cna.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "patientToSamplesDict = {}\n",
    "for sample in df_cna.columns[1:]:\n",
    "    patient = \"-\".join(sample.split('-')[0:2])\n",
    "    if patient not in patientToSamplesDict:\n",
    "        patientToSamplesDict[patient] = set()\n",
    "    patientToSamplesDict[patient].add(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AMER1',\n",
       " 'AR',\n",
       " 'ARAF',\n",
       " 'ATRX',\n",
       " 'BCOR',\n",
       " 'BTK',\n",
       " 'CRLF2',\n",
       " 'EIF1AX',\n",
       " 'FAM58A',\n",
       " 'GATA1',\n",
       " 'KDM5C',\n",
       " 'KDM6A',\n",
       " 'MED12',\n",
       " 'RBM10',\n",
       " 'SH2D1A',\n",
       " 'STAG2',\n",
       " 'XIAP',\n",
       " 'ZRSR2'}"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_snv[df_snv['Chromosome'] == 'X'].Hugo_Symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "male_patients = set([\"P-0002574\", \"P-0000281\", \"P-0000281\",\n",
    "                     \"P-0015988\", \"P-0005904\", \"P-0000694\",\n",
    "                     \"P-0001354\", \"P-0007684\", \"P-0000423\",\n",
    "                     \"P-0007700\", \"P-0016667\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_frequencies(df_snvs, df_cnas, patient, snvs):\n",
    "    nrSamples = len(patientToSamplesDict[patient])\n",
    "    sampleList = list(patientToSamplesDict[patient])\n",
    "    nrSNVs = len(snvs)\n",
    "#     print(patient, snvs, nrSNVs)\n",
    "    with open(\"%s.tsv\" % patient, \"w\") as f:\n",
    "        f.write(\"%d #anatomical sites\\n\" % 1)\n",
    "        f.write(\"%d #samples\\n\" % nrSamples)\n",
    "        f.write(\"%d #characters\\n\" % nrSNVs)\n",
    "        f.write(\"\\t\".join([\"#sample_index\", \"sample_label\", \n",
    "                 \"anatomical_site_index\", \"anatomical_site_label\", \n",
    "                 \"character_index\", \"character_label\", \n",
    "                 \"f-\", \"f+\\n\"]))\n",
    "        for sampleIdx, sample in enumerate(sampleList):\n",
    "            for snvIdx, snv in enumerate(snvs):\n",
    "                s = snv.split(\":\")\n",
    "                symbol = s[2]\n",
    "                chr = s[0]\n",
    "                pos = int(s[1])\n",
    "                \n",
    "#                 print(sample, symbol, chr, pos)\n",
    "#                 print(df_snvs[(df_snvs[\"Tumor_Sample_Barcode\"] == sample)\n",
    "#                                      & (df_snvs[\"Hugo_Symbol\"] == symbol)\n",
    "#                                      & (df_snvs[\"Chromosome\"] == chr)\n",
    "#                                      & (df_snvs[\"Start_Position\"] == pos)\n",
    "#                                     ][\"t_alt_count\"])\n",
    "\n",
    "                try:\n",
    "                    var = float(df_snvs[(df_snvs[\"Tumor_Sample_Barcode\"] == sample)\n",
    "                                         & (df_snvs[\"Hugo_Symbol\"] == symbol)\n",
    "                                         & (df_snvs[\"Chromosome\"] == chr)\n",
    "                                         & (df_snvs[\"Start_Position\"] == pos)\n",
    "                                        ][\"t_alt_count\"])\n",
    "                    tot = float(df_snvs[(df_snvs[\"Tumor_Sample_Barcode\"] == sample)\n",
    "                                         & (df_snvs[\"Hugo_Symbol\"] == symbol)\n",
    "                                         & (df_snvs[\"Chromosome\"] == chr)\n",
    "                                         & (df_snvs[\"Start_Position\"] == pos)\n",
    "                                        ][\"t_depth\"])\n",
    "                    vaf = var/tot\n",
    "                except:\n",
    "                    vaf = 0.\n",
    "            \n",
    "                f.write(\"\\t\".join(map(str, [sampleIdx, sample, 0, \"P\", snvIdx, snv, vaf, vaf])) + \"\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_spruce_input(df_snvs, df_cnas, patientToSamplesDict, male_patients):\n",
    "    for patient in patientToSamplesDict:\n",
    "        if patient not in male_patients:\n",
    "            snv_candidates = set()\n",
    "            for sample in patientToSamplesDict[patient]:\n",
    "                # determine whether SNV is copy neutral in all tumor samples\n",
    "                for idx, row in df_snvs[df_snvs[\"Tumor_Sample_Barcode\"] == sample].iterrows():\n",
    "                    classification = row[\"Variant_Classification\"]\n",
    "                    symbol = row[\"Hugo_Symbol\"]\n",
    "                    chr = row[\"Chromosome\"]\n",
    "                    pos = row[\"Start_Position\"]\n",
    "#                     if classification in ['Missense_Mutation', 'Nonsense_Mutation', 'Nonstop_Mutation', 'Splice_Site', 'Translation_Start_Site']:\n",
    "                    if classification not in ['Silent', 'Intron']:\n",
    "                        snv_candidates.add(\"%s:%s:%s\" % (chr, pos, symbol))\n",
    "            snvs = set()\n",
    "            for snv in snv_candidates:\n",
    "                s = snv.split(\":\")\n",
    "                symbol = s[2]\n",
    "                chr = s[0]\n",
    "                pos = int(s[1])\n",
    "\n",
    "                neutral = True\n",
    "                for sample in patientToSamplesDict[patient]:\n",
    "                    if sample in df_cnas.columns:\n",
    "                        if len(df_cnas[df_cnas[\"Hugo_Symbol\"] == symbol][sample]) == 0:\n",
    "                            neutral = False\n",
    "                        else:\n",
    "                            copynumber = df_cnas[df_cnas[\"Hugo_Symbol\"] == symbol][sample].iloc[0]\n",
    "                            if copynumber != 0:\n",
    "                                neutral = False\n",
    "                if neutral:\n",
    "                    snvs.add(snv)\n",
    "            generate_frequencies(df_snvs, df_cnas, patient, snvs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>VAF</th>\n",
       "      <th>TCN</th>\n",
       "      <th>LCN</th>\n",
       "      <th>CCF</th>\n",
       "      <th>CLONALITY</th>\n",
       "      <th>Total_Index_Lesions_In_Patient</th>\n",
       "      <th>Num_Other_Mets_In_Patient</th>\n",
       "      <th>CLONALITY_In_Other_Mets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6726</th>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>6597</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>11143994</td>\n",
       "      <td>11143994</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "6726     SMARCA4            6597  MSKCC     GRCh37         19        11143994   \n",
       "\n",
       "      End_Position Strand Variant_Classification Variant_Type  ...  \\\n",
       "6726      11143994      +      Missense_Mutation          SNP  ...   \n",
       "\n",
       "     HIGH_INF_POS MOTIF_SCORE_CHANGE  VAF TCN  LCN CCF CLONALITY  \\\n",
       "6726          NaN                NaN  NaN NaN  NaN NaN       NaN   \n",
       "\n",
       "     Total_Index_Lesions_In_Patient Num_Other_Mets_In_Patient  \\\n",
       "6726                            NaN                       NaN   \n",
       "\n",
       "      CLONALITY_In_Other_Mets  \n",
       "6726                      NaN  \n",
       "\n",
       "[1 rows x 96 columns]"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_snv[(df_snv[\"Tumor_Sample_Barcode\"] == \"P-0003381-T01-IM5\")\n",
    "       & (df_snv[\"Hugo_Symbol\"] == \"SMARCA4\")\n",
    "       & (df_snv[\"Chromosome\"] == \"19\")\n",
    "#        & (df_snv[\"Start_Position\"] == 11143994)\n",
    "      ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "generate_spruce_input(df_snv, df_cna, patientToSamplesDict, male_patients)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{\"5'UTR\",\n",
       " 'Frame_Shift_Del',\n",
       " 'Frame_Shift_Ins',\n",
       " 'In_Frame_Del',\n",
       " 'In_Frame_Ins',\n",
       " 'Intron',\n",
       " 'Missense_Mutation',\n",
       " 'Nonsense_Mutation',\n",
       " 'Nonstop_Mutation',\n",
       " 'Silent',\n",
       " 'Splice_Site',\n",
       " 'Translation_Start_Site'}"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df_snv[\"Variant_Classification\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'P-0011355-CMO-P', 'P-0011355-T01-IM5'}"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patientToSamplesDict[\"P-0011355\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
